### ISSUE TITLE
Timestamps are not being created when using $setOnInsert

### ISSUE THREAD

**Issue by lbertoncello** [Automattic/mongoose#10460]
Maybe I've found a new bug related to timestamps, like [this one](https://github.com/Automattic/mongoose/issues/10447) I've described recently.

Timestamps are not being created when I use `$setOnInsert` in `findOneAndUpdate`.

You can see here the code I've used to test it:

```
const mongoose = require('mongoose');
const { Schema } = mongoose;

const userSchema = new Schema({
	email: {
		type: String,
		required: true,
	},
	fullname: {
		type: String,
		required: true,
	},
	dateOfBirth: {
		type: Date,
		required: true,
	},
	contacted: {
		type: Boolean,
		required: true,
		default: false,
	},
}, {
	timestamps: true,
});

const eventSchema = new Schema({
	code: {
		type: String,
		required: true,
	},
	template: String,
	users: {
		type: [ userSchema ],
		required: false,
	},
	message: String,
}, {
	timestamps: true,
});


const churchSchema = new Schema({
	churchId: {
		type: Number,
		required: true,
	},
	integration: String,
	service: String,
	events: {
		type: [ eventSchema ],
		required: false,
	},
}, {
	timestamps: true,
});

const Test = mongoose.model('Church', churchSchema);

async function run () {
	await mongoose.connect('mongodb://localhost:27017/test', {
		useNewUrlParser: true,
		useUnifiedTopology: true,
	});
	await mongoose.connection.dropDatabase();

	const query = { churchId: 1 };
	const update = {
		$setOnInsert: {
			churchId: 1,
			integration: 'Hello',
			service: 'world',
			events: [
				{
					code: 'A',
					template: 'B',
					users: [
						{
							email: 'Test@testerson.com',
							fullname: 'Fullname',
							dateOfBirth: new Date(),
							contacted: true,
						},
					],
					message: 'message',
				},
			],
		},
	};
	const options = {
		upsert: true,
		new: true,
		timestamps: true,
	};

	const entry = await Test.findOneAndUpdate(query, update, options);

	const result = await Test.findOneAndUpdate({
		_id: entry._id,
		events: { $elemMatch: { code: 'A' } },
	}, {
		$addToSet: {
			'events.$.users': {
				email: 'randal@test.com',
				fullname: 'fred',
				dateOfBirth: new Date(),
				contacted: true,
			},
		},
	}, {
		upsert: true,
		new: true,
		timestamps: true,
	});

	console.log(result);
	console.log(result.events[0].users);
}

run();
```

I've got the following output:

```
{
  _id: 60f1d8db995dbbbe6da2aa4a,
  churchId: 1,
  __v: 0,
  createdAt: 2021-07-16T19:07:07.281Z,
  events: [
    {
      _id: 60f1d8db6b5880242c18fce6,
      code: 'A',
      template: 'B',
      users: [Array],
      message: 'message'
    }
  ],
  integration: 'Hello',
  service: 'world',
  updatedAt: 2021-07-16T19:07:07.303Z
}
[
  {
    contacted: true,
    _id: 60f1d8db6b5880242c18fce7,
    email: 'Test@testerson.com',
    fullname: 'Fullname',
    dateOfBirth: 2021-07-16T19:07:07.279Z
  },
  {
    contacted: true,
    _id: 60f1d8db6b5880242c18fce9,
    email: 'randal@test.com',
    fullname: 'fred',
    dateOfBirth: 2021-07-16T19:07:07.303Z,
    updatedAt: 2021-07-16T19:07:07.303Z,
    createdAt: 2021-07-16T19:07:07.303Z
  }
]
```

Is this a bug or an expected behaviour?
___
MongoDB: 4.4.6
Mongoose: 5.13.3
Node.js: v14.17.0

**Comment by vkarpov15** [comment]
This is a bug. The fix will be in v5.13.4 :+1:

**Comment by lbertoncello** [comment]
> This is a bug. The fix will be in v5.13.4 👍

Ok, thanks!
